lms/djangoapps/oauth_dispatch/views.py

```diff
@@ -72,6 +82,71 @@ class AccessTokenView(_DispatchingView):
     dot_view = dot_views.TokenView
     dop_view = dop_views.AccessTokenView
 
+    @cached_property
+    def claim_handlers(self):
+        """ Returns a dictionary mapping scopes to methods that will add claims to the JWT payload. """
+
+        return {
+            'email': self._attach_email_claim,
+            'profile': self._attach_profile_claim
+        }
+
+    def dispatch(self, request, *args, **kwargs):
+        response = super(AccessTokenView, self).dispatch(request, *args, **kwargs)
+
+        if response.status_code == 200:
+            content = json.loads(response.content)
+            token = self._generate_jwt(request, content)
+            content['access_token'] = token
+            response.content = json.dumps(content)
+
+        return response
+
+    def _generate_jwt(self, request, content):
+        """ Returns a JWT access token. """
+
+        access_token = content['access_token']
+        access_token_obj = self.get_adapter(request).get_access_token(access_token)
+        user = access_token_obj.user
+        now = int(time())
+        content['token_type'] = 'JWT'
+
+        scope = content['scope']
+        payload = {
+            'iss': settings.JWT_AUTH['JWT_ISSUER'],
+            'aud': settings.JWT_AUTH['JWT_AUDIENCE'],
+            'jti': access_token,
+            'exp': now + content['expires_in'],
+            'iat': now,
+            'preferred_username': user.username,
+            'scope': scope,
+        }
+
+        scopes = scope.split(' ')
```

Comment:
Are the scopes always space-delimited?  Is there a constant somewhere that specifies what the delimiter is?